### PR TITLE
CTP-2244 code review

### PR DESCRIPTION
**Pull request by aydevworks**
Related to Jira story: https://ucldata.atlassian.net/browse/CTP-2244

Overall the code was good to me. I would suggest some minor changes as I commented below.
May be add a README.md to describe what this plugin does?

### CODE REVIEW

File: block_alerts.php [hunk 1]
```diff
@@ -29,19 +29,18 @@ class block_alerts extends block_base {
      * @return void
      */
     public function init() {
-        global $USER;
```
Comment (aydevworks) [inline block_alerts.php:32]: It's not used, so could be removed.

File: block_alerts.php [hunk 2]
```diff
@@ -29,19 +29,18 @@ class block_alerts extends block_base {
...
         $this->title = get_string('pluginname', 'block_alerts');
     }
 
     /**
      * Gets the block settings.
      *
+     * @return void
      */
     public function specialization() {
-        if (isset($this->config->title)) {
```
Comment (aydevworks) [inline block_alerts.php:41]: `isset($this->config->title)` is set when the value is an empty string. e.g. when user add the title and then later removed it.
I think it's better to check for empty string using `empty()`.

File: block_alerts.php [hunk 3]
```diff
@@ -125,10 +123,10 @@ public function fetch_alerts() : array {
             $template->alerts[] = $alerts;
         }
 
-        // Set first element as active for carosel version.
```
Comment (aydevworks) [inline block_alerts.php:128]: Spelling (carousel)

File: version.php [hunk 4]
```diff
@@ -25,6 +25,7 @@
 defined('MOODLE_INTERNAL') || die();
 
 $plugin->version   = 2023062600;        // The current plugin version (Date: YYYYMMDDXX).
+$plugin->release   = '1.0';
```
Comment (aydevworks) [inline version.php:28]: Just looks more completed version.php 😄 
https://docs.moodle.org/dev/version.php

File: block_alerts.php (hunk 2 above)
Comment (aydevworks) [reply block_alerts.php:41]: `else {
            $this->title = get_string('pluginname', 'block_alerts');
    }`
this is not necessary as it is set in the `init()` already.

File: version.php (hunk 4 above)
Comment (aydevworks) [inline version.php:28]: Just looks more completed version.php 😄 
https://docs.moodle.org/dev/version.php

File: block_alerts.php (hunk 2 above)
Comment (stuartlamour) [reply block_alerts.php:41]: Allow for no title looks good when tested locally - thanks:)

**Review by stuartlamour — APPROVED** [review]
Thanks @aydevworks

**Pull request merged**